tools/codegen/api/cpp.py

```diff
@@ -158,6 +158,26 @@ def returns_type(rs: Sequence[Return]) -> str:
 def default_expr(d: str, t: Type) -> str:
     if d == 'None' and str(t) == 'Tensor?':
         return '{}'
+    if isinstance(t, BaseType) and t.name is BaseTy.str:
+        # Schema allows single quotes but C++ needs double
+        if len(d) >= 2 and d[0] == "'" and d[-1] == "'":
+            s = ''
+            i = 1
+            while i + 1 < len(d):
```

Comment:
This chunk of code is what I would expect to see in C, but not in Python. Manual indexing is verbose, and holds up poorly when you need to add extensions in the future. In Python, you should just parse the type into an actual string, and then escape it for double quotes (which would just be a simple `replace('"', '\\"')`)

Reply:
(Parsing can be done with `ast.literal_eval`, if we define the permissible string format to be all of Python (which should be the case, given that we support single quotes)

Reply:
Or, better yet, we can disallow single-quoted strings and then define the string format to be exactly JSON string format.

Reply:
I am only allowing single quotes because the JIT frontend also allows it. I can remove this if not matching the JIT frontend exactly is okay.